File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    int maxDistance(vector<int>& A, vector<int>& B) {
        int n1 = A.size(), n2 = B.size();
        int ptr2 = 0;

        int result = 0;

        for(int i = 0; i < n1; i++) {
            auto it = lower_bound(B.rbegin(), B.rend(), A[i]);
            int pos = n2 - distance(B.rbegin() , it) - 1;
            if (pos > i) result = max(result, pos - i);
        }
        return result;
    }
};

File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        ans = 0
        n1, n2 = len(nums1), len(nums2)
        for i in range(n1):
            l, r = 0, n2-1
            while l < r:
                mid = (l+r+1)//2
                if nums2[mid] >= nums1[i]:
                    l = mid
                else:
                    r = mid-1
            ans = max(ans, l-i)
        return ans